geopandas/io/tests/test_file.py

```diff
@@ -232,7 +234,63 @@ def test_to_file_datetime(tmpdir, driver, ext, time, engine):
         assert_series_equal(df["b"], df_read["b"])
 
 
-def test_read_file_mixed_datetimes(tmpdir):
+dt_exts = ["gpkg", "geojson"]
+
+
+def write_invalid_date_file(date_str, tmpdir, ext, engine):
+    tempfilename = os.path.join(str(tmpdir), f"test_invalid_datetime.{ext}")
+    df = GeoDataFrame(
+        {
+            "date": ["2014-08-26T10:01:23", "2014-08-26T10:01:23", date_str],
+            "geometry": [Point(1, 1), Point(1, 1), Point(1, 1)],
+        }
+    )
+    # Schema not required for GeoJSON since not typed, but needed for GPKG
+    if ext == "geojson":
+        df.to_file(tempfilename)
+    else:
+        schema = {"geometry": "Point", "properties": {"date": "datetime"}}
+        if engine == "pyogrio" and not fiona:
+            # (use schema to write the invalid date without pandas datetimes
```

Comment:
To test this with pyogrio, we will probably have to use the "raw" write functionality of pyogrio (not starting from a pandas dataframe, but from numpy arrays, and then we can have a datetime64[s] dtype array to represent the out of bounds values). At least that should work for the valid but out of bounds datetime, still not for the invalid datetime.

Reply:
In addition, when using numpy (or pandas) we could also test with an explicit "NaT" in the values being written.

Reply:
> To test this with pyogrio, we will probably have to use the "raw" write functionality of pyogrio (not starting from a pandas dataframe, but from numpy arrays, and then we can have a datetime64[s] dtype array to represent the out of bounds values). At least that should work for the valid but out of bounds datetime, still not for the invalid datetime.

That makes sense, I thought this was the simplest compromise for now - we'll still at least test with pyogrio for the environments with both fiona and pyogrio installed.